subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultTemporaryFileProvider.java

```diff
@@ -44,7 +44,9 @@ public File createTemporaryFile(String prefix, @Nullable String suffix, String..
         File dir = new File(baseDirFactory.create(), CollectionUtils.join("/", path));
         GFileUtils.mkdirs(dir);
         try {
-            return File.createTempFile(prefix, suffix, dir);
+            File tempFile = File.createTempFile(prefix, suffix, dir);
+            tempFile.deleteOnExit();
```

Comment:
should we delete any empty dir this leaved because of https://github.com/gradle/gradle/pull/9777/files#diff-351eea53344ac3ec211c189569aaa31dR45 ?